### PR TITLE
Vendor dependencies, update k8s to 1.16.3

### PR DESCRIPTION
**Pull request by sslavic**
Towards giantswarm/giantswarm#7675

### CODE REVIEW

**Comment by tomahawk28** [comment]
@sslavic Implmentation changed due to tenantcluster changes, can you maybe take a look? cc> @xh3b4sd

File: resource.go [hunk 1]
```diff
@@ -26,8 +26,13 @@ type ResourceConfig struct {
 	ClusterEndpointFunc func(v interface{}) (string, error)
 	ClusterIDFunc       func(v interface{}) (string, error)
 	ClusterStatusFunc   func(v interface{}) (providerv1alpha1.StatusCluster, error)
-	Logger              micrologger.Logger
-	NodeCountFunc       func(v interface{}) (int, error)
+	// TODO replace this with a G8sClient to fetch the node versions from the
```
Comment (tomahawk28) [inline resource.go]: @xh3b4sd I saw you posted this TODO, still valid?

**Review by tomahawk28 — APPROVED** [review]

**Comment by tomahawk28** [comment]
Okay, It was meant to delete tenantcluster usage in here, let's go with this.

File: resource.go (hunk 1 above)
Comment (xh3b4sd) [reply resource.go]: I think meanwhile we are moving into a direction where `statusresource` just not be a thing anymore in the future. Let's not do too much in here.

**Comment by sslavic** [comment]
> Okay, It was meant to delete tenantcluster usage in here, let's go with this.

@tomahawk28 I think this was not what was meant, or at least it doesn't make sense to me. IMO using configured `TenantCluster`, `StatusResource` is supposed to get `K8sClient` with something like https://github.com/giantswarm/node-operator/pull/100/files#diff-7eeb593479b32373553c6330a7bcc34e

The way it is after this PR, there's no way to construct `K8sClient` and configure it to `StatusResource` e.g. in azure-operator's https://github.com/giantswarm/azure-operator/blob/master/service/controller/v9/resource_set.go#L127-L144 - cluster ID and endpoint are not know there, just function is passed for extracting them.

Please consider reintroducing `TenantCluster` in the configuration, and using it in `StatusResource` as in the example in `node-operator`. It would be great if @xh3b4sd and @rossf7 would review/verify.

**Comment by tomahawk28** [comment]
@sslavic Got it, Thanks to pointing node-operator, I will update statusresource after resolving the issue in chart-operator.

**Pull request merged**